docs/fundamentals/networking/telemetry/metrics.md

```diff
@@ -0,0 +1,157 @@
+
+### Example app
+
+For the sake of this tutorial, let's create a simple app that sends HTTP requests to various endpoints in parallel.
+
+```dotnetcli
+dotnet new console -o HelloBuiltinMetrics
+cd ..\HelloBuiltinMetrics
+```
+
+Replace the contents of `Program.cs` with the following sample code:
+
+:::code language="csharp" source="snippets/metrics/Program.cs" id="snippet_1":::
+
+### View metrics with dotnet-counters
+
+[`dotnet-counters`](../../../core/diagnostics/dotnet-counters.md) is a cross-platform performance monitoring tool for ad-hoc health monitoring and first-level performance investigation.
+
+```dotnetcli
+dotnet tool install --global dotnet-counters
+```
+
+When running against a .NET 8.0+ process, it will enable the instruments defined by the `--counters` argument and display the measurements, continuosuly refreshing the console with the latest numbers:
+
+```console
+ dotnet-counters monitor --counters System.Net.Http,System.Net.NameResolution -n HelloBuiltinMetrics
+```
+
+### View metrics in Grafana with OpenTelemetry and Prometheus
+
+#### Overview
+
+[OpenTelemetry](https://opentelemetry.io/):
+
+- Is a vendor-neutral open-source project supported by the [Cloud Native Computing Foundation](https://www.cncf.io/).
+- Standardizes generating and collecting telemetry for cloud-native software.
+- Works with .NET using the .NET metric APIs.
+- Is endorsed by [Azure Monitor](/azure/azure-monitor/app/opentelemetry-overview) and many APM vendors.
+
+This tutorial shows one of the integrations available for OpenTelemetry metrics using the OSS [Prometheus](https://prometheus.io/) and [Grafana](https://grafana.com/) projects. The metrics data flow:
+
+1. The .NET metric APIs record measurements from the example app.
+1. The OpenTelemetry library running in the app aggregates the measurements.
+1. The Prometheus exporter library makes the aggregated data available via an HTTP metrics endpoint. 'Exporter' is what OpenTelemetry calls the libraries that transmit telemetry to vendor-specific backends.
+1. A Prometheus server:
+
+   - Polls the metrics endpoint
+   - Reads the data
+   - Stores the data in a database for long-term persistence. Prometheus refers to reading and storing data as *scraping* an endpoint.
+   - Can run on a different machine
+
+1. The Grafana server:
+
+   - Queries the data stored in Prometheus and displays it on a web-based monitoring dashboard.
+   - Can run on a different machine.
+
+#### Configure the example app to use OpenTelemetry's Prometheus exporter
+
+Add a reference to the OpenTelemetry Prometheus exporter to the example app:
+
+```dotnetcli
+dotnet add package OpenTelemetry.Exporter.Prometheus.HttpListener --prerelease
+```
+
+> [!NOTE]
+> This tutorial uses a pre-release build of OpenTelemetry's Prometheus support available at the time of writing.
+
+Update `Program.cs` with OpenTelemetry configuration:
```

Comment:
Should we leave a note that this configuration will differ for ASP.NET Core apps (e.g. no `HttpListener`) and point readers there?